packages/@expo/config/src/getConfig.ts

```diff
@@ -36,6 +36,15 @@ export function getDynamicConfig(configPath: string, request: ConfigContext): Dy
 export function getStaticConfig(configPath: string): AppJSONConfig | ExpoConfig {
   const config = JsonFile.read(configPath, { json5: true });
   if (config) {
+    // We check _hasBaseStaticConfig later when reading the dynamic config to see if
+    // there is a static config that is not inherited from the dynamic config.
+    if (config.expo) {
+      // @ts-ignore
+      config.expo._hasBaseStaticConfig = true;
```

Comment:
Move this to be right before we merge with dynamic, then delete it afterwards so there are no side-effects. 

Maybe even use a Symbol instead of a normal key.

Reply:
@EvanBacon I refactored this to be internal to `resolveConfigExport()` with the psuedo-private key, but I can't seem to get it to work with a `Symbol`. What's strange is I can definitely read the symbol when spreading the input, but after it comes out of the dynamic config function, the symbol is `undefined`, even though all that should be happening is that it's being spread. Any ideas?

Test code:
```
// demo that spreading the original config is carrying over the symbol...
  console.log(
    'request?.config?.[hasBaseStaticConfig]',
    // @ts-ignore
    { ...request?.config }[hasBaseStaticConfig]
  );
  // ... but the dynamic config output is not
  console.log('result?.[hasBaseStaticConfig]', result?.[hasBaseStaticConfig]);
  ```
  
  Console output (with an app.config.js that returns the `...config`):
  ```
  request?.config?.[hasBaseStaticConfig] true
result?.[hasBaseStaticConfig] undefined
```